### PR TITLE
Add SyncManager to Web5 Managed Agent

### PR DESCRIPTION
**Pull request by frankhinek**
This PR will:
- Re-introduce [this sync implementation](https://github.com/TBD54566975/web5-js/blob/main/packages/old/web5-user-agent/src/sync-api.ts) to restore sync capability.
- Resolves an issue with the original sync implementation with cascading echoes that would occur due to random lexicographical ordering of sync jobs.  Unresolved is a single echo after a pull operation, which is tracked in Issue #207.
- Resolves Issue #95.
- Previously sync had no tests so basic tests have been added for `push()` and `pull()`.  Testing for more scenarios and edge cases is needed, which is tracked in Issue #206
- Temporarily adds utility methods to `DwnManager` that should be refactored as tracked in Issue #205.
- Adds `SyncManager` to the tests / test agents.
- Improves test coverage in the `@web5/agent` package for `DwnManager` handling `EventsGet`, `MessagesGet`, `ProtocolsConfigure`, and `ProtocolsQuery`.
- Adds test coverage in the `@web5/api` package for `Protocol`.

### CODE REVIEW

File: packages/agent/src/dwn-manager.ts [hunk 1]
```diff
@@ -532,4 +532,118 @@ export class DwnManager {
...
+
+
+
+
+
+
+
+
+
+
```
Comment (mistermoe) [inline packages/agent/src/dwn-manager.ts:560]: adding these newlines to create some distance between what we intend to keep around vs. the stuff we hope to refactor?

**Review by mistermoe — APPROVED** [review]
🥇

File: packages/agent/src/dwn-manager.ts (hunk 1 above)
Comment (frankhinek) [reply packages/agent/src/dwn-manager.ts:560]: Yes, that's the only reason they are there -- as a reminder to get rid of this patchwork of methods when we refactor.
